src/Makefile

```diff
@@ -60,7 +61,7 @@ ifeq ($(shell uname), Darwin)
 else
 ifeq ($(shell uname), Linux)
 	#$(warning the following command will probably fail, in that case add -fPIC to your CXXFLAGS and remake all)
-	$(CXX) -shared -o $(KALDILIBDIR)/libkaldi.so -Wl,-soname=libkaldi.so,--whole-archive  $(SUBDIRS_LIB:=/kaldi-*.a) -Wl,--no-whole-archive
+	$(CXX) -shared -o $(KALDILIBDIR)/$(KALDI_SONAME) -Wl,-soname=$(KALDI_SONAME),--whole-archive  $(SUBDIRS_LIB:=/kaldi-*.a) $(LDLIBS) -Wl,--no-whole-archive
```

Comment:
yes, it makes it so the .so links to others (e.g. `liblapack.so` etc). e.g. i was building like this:

```bash
$ OPENFST_VER=1.4.1 ./configure --fst-root=/usr/local --openblas-root=/usr --mathlib=OPENBLAS --shared
$ EXTRA_LDLIBS="-llapack" KALDI_SONAME="libkaldiXXX.so" $(MAKE) biglib
```

and without that change the kaldi so did no link to these other libs: 
```bash
$ ldd /usr/local/lib/libkaldi59e4691.so
	linux-vdso.so.1 =>  (0x00007ffe0e1fc000)
	liblapack.so.3 => /usr/lib/liblapack.so.3 (0x00007f4bddd0a000)
	libfst.so.3 => /usr/local/lib/libfst.so.3 (0x00007f4bdd839000)
	libopenblas.so.0 => /usr/lib/libopenblas.so.0 (0x00007f4bdc02f000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f4bdbe11000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f4bdbc0d000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f4bdb8fa000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f4bdb5f4000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f4bdb3dd000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f4bdb017000)
	libblas.so.3 => /usr/lib/libblas.so.3 (0x00007f4bd980e000)
	libgfortran.so.3 => /usr/lib/x86_64-linux-gnu/libgfortran.so.3 (0x00007f4bd94e7000)
	/lib64/ld-linux-x86-64.so.2 (0x000056137726f000)
	libquadmath.so.0 => /usr/lib/x86_64-linux-gnu/libquadmath.so.0 (0x00007f4bd92a7000)
```

i can make a separate change for that if you prefer.